### PR TITLE
Fix issue #285: Multi-line descriptions in messages on resolved issues are not properly parsed

### LINKED ISSUE #285: Multi-line descriptions in messages on resolved issues are not properly parsed
It seems that when an issue has been resolved, multi-line descriptions of the work done are not properly parsed.

For example, I saw one like the following:

```
This pull request fixes https://github.com/All-Hands-AI/openhands-resolver/issues/283.

The PR successfully addressed the issue by:

Automatic fix generated by [OpenHands](https://github.com/All-Hands-AI/OpenHands/) 🙌
```

presumably, there should be something else after "The PR successfully addressed the issue by:", such as a bullet list.

This probably needs to be fixed when we parse the output of the `guess_success()` function in `resolve_issues.py`, specifically for issue resolution (not PR updates).

### PR DESCRIPTION
**Pull request by openhands-agent**
This pull request fixes #285.

The issue has been successfully resolved. The PR addressed the core problem by:

Automatic fix generated by [OpenHands](https://github.com/All-Hands-AI/OpenHands/) 🙌

### CODE REVIEW

**Review by neubig — APPROVED** [review]
LGTM, thanks!

**Review by neubig — CHANGES_REQUESTED** [review]
Actually, pre-commit is failing. Please make sure that pre-commit and tests both pass.

**Comment by openhands-agent** [comment]
New OpenHands update

File: openhands_resolver/issue_definitions.py [hunk 1]
```diff
@@ -168,15 +168,17 @@ def guess_success(self, issue: GithubIssue, history: list[Event], llm_config: LL
         {last_message}
 
         (1) has the issue been successfully resolved?
-        (2) If the issue has been resolved, please provide an explanation of what was done in the PR that can be sent to a human reviewer on github. If the issue has not been resolved, please provide an explanation of why.
+        (2) If the issue has been resolved, please provide an explanation of what was done in the PR that can be sent to a human reviewer on github. The explanation should be a bullet-point list of changes made, with each point starting with "- ". If the issue has not been resolved, please provide an explanation of why.
```
Comment (neubig) [inline openhands_resolver/issue_definitions.py:171]:
```suggestion
        (2) If the issue has been resolved, please provide an explanation of what was done in the PR that can be sent to a human reviewer on github. If the issue has not been resolved, please provide an explanation of why.
```

File: openhands_resolver/issue_definitions.py [hunk 2]
```diff
@@ -176,7 +176,9 @@ def guess_success(self, issue: GithubIssue, history: list[Event], llm_config: LL
         true/false
 
         --- explanation
-        ...
+        - Change 1
+        - Change 2
+        - ...
```
Comment (neubig) [inline openhands_resolver/issue_definitions.py:181]:
```suggestion
        ...
```

**Review by neubig — APPROVED** [review]
LGTM now.

**Pull request merged**